### DISCUSSION TITLE
Help with terraria server using raspberry pi (using TShock)

### DISCUSSION THREAD

**Discussion by Dedave2** [Pryaxis/TShock#2241] [Q&A]
I am using raspberry pi to make a server I really need help. So, I downloaded and extracted tshock on the raspberry pi and everything to downloads (the default directory it extracts files to) and then I tried doing mono ./TerrariaServer.exe (i did do the cd command in downloads folder) and for some reason it says "No such file or directory". I've tried everything, looked everywhere for solutions, nothing. Please help.

Also I wanna know, I want to use this to host a server for me and my friend in terraria and we already have a world I just want it to be 24/7 since we have different timezones. Is there a way to add a world that we have been already playing on? I have the file of the world and everything I just don't know how to insert it at all or what to do with it using a console

Btw i used this tutorial https://www.instructables.com/Raspberry-Pi-Terraria-Server/

And this tutorial https://gist.github.com/BalzGuenat/65c248d03895b5a030ba

Please explain everything to me like for noob because i am very much noob i dont know anything about coding or anything i just followed a tutorial and i got my raspberry pi today

If you help me I will appreciate it so so much

**Comment by Thunder1723** [comment]
I'm also new so I'm not sure how much I can help but what is your directory and where is it saved?  For example I saved the Tshock folder to my desktop and the directory command I use is 

cd /home/pi/Desktop/Tshock4.4.0_Pre15_Terraria1.4.1.2

As long as you leave your pi running with a reliable connection, I highly recommend plugging into an Ethernet port, just leave the server open and don't close the terminal.  If you are also going to leave it on all the time it is completely necessary to invest in some active cooling (heat sink and fan) to run a server.  Even if you don't overclock the pi running it 24/7 will be taxing and it will die much faster without good cooling.  If you want advice on overclocking I would be glad to tell you what I have learned but you didn't ask about that.  

Inserting the Terraria world files you already have is a crap shoot because its kind of hidden.  The only world file you need is the one that ends in .wld.  The other files are backups and you don't need them to load up the world.  The directory that my Terraria worlds are stored in is

/home/pi/.local/share/

Look through that using the pi's file manager.  If you get to the right spot there will be a folder labeled Terraria and that's where world files are stored and accessed by Tshock.  I use windows xrdp to connect to my pi so I can copy and paste things back and forth but using a USB stick is also a fairly simple task in my experience so far.

**Reply by Dedave2** [reply]
The folder is in downloads. I also did that command, but with downloads instead of desktop

**Reply by Dedave2** [reply]
Also, the sevrer is just for me andmy friend and most of the time only 1 person will be on it so i dont think i need to cool it lol

**Reply by Thunder1723** [reply]
You definitely need to cool it, when I ran my pi with just one of my friends on my server it would get up to 50 C with cooling from a cana kit.  I can't imagine how hot a pi with no active cooling would get.  Its not necessarily about keeping it under the thermal throttle limit its about extending the life of the pi by keeping it cooler.  Its worth it to spend $10 - $30 to extend the life of your pi a couple extra years.

If the command still isn't working I would try moving the file folder to desktop just to see what happens.  You can also delete it and re download it as that fixes it for some people.
